### ISSUE TITLE
Create trivy plugin to query on trivy.db

### ISSUE THREAD

**Issue by alfeilex** [aquasecurity/trivy-db#454]
Hello,

I would like to ask if it's possible to create a trivy plugin that uses the query components on the trivy.db to search for specific CVE's that match a vendor, or product and version. Is there an API to using the database for other languages like Java?

Finally, we want to include trivy in our CLI tool, but we are looking for a functionality so that the user can search for a CVE given the vendor/product and version. The output for that query should include matching CVE's and perhaps references/fixed versions and severity.

Thank you

**Comment by knqyf263** [comment]
https://github.com/aquasecurity/trivy/discussions/7740

**Comment by alfeilex** [comment]
The answer of the issue in trivy did not help me. Can you please reopen the issue? I want to know if there is an API in trivy-db. Boltwiz does not help me, because it's not possible to integrate it in the CLI

**Comment by knqyf263** [comment]
[bbolt](https://github.com/etcd-io/bbolt) has a CLI.

**Comment by alfeilex** [comment]
Let me explain my problem in more detail:

We have developed a CLI based on Java, where we offer various tools. We want to provide our users with the ability to detect CVEs (Common Vulnerabilities and Exposures) for the tools and their versions that we offer.

We have tested Trivy and would like to integrate it into our product. However, the issue is that there is no query that returns CVEs based solely on vendor and version. Therefore, the idea is to write a plugin for Trivy that performs such a query using your database. Hence, my question is whether there are interfaces (APIs) available that we can use for this purpose.

We would like to avoid using another external tool like bbolt or boltwiz for now, as it is based on Go (as far as I know), and we are focused on Java.
